parser/parser_table.go

```diff
@@ -782,14 +795,127 @@ func (p *Parser) parseOrderExpr(pos Pos) (*OrderExpr, error) {
 		direction = OrderDirectionDesc
 		_ = p.lexer.consumeToken()
 	}
+
+	// Parse optional WITH FILL clause
+	var fill *Fill
+	if p.matchKeyword(KeywordWith) {
+		savedState := p.lexer.saveState()
+		_ = p.lexer.consumeToken() // consume WITH
+		if p.matchKeyword(KeywordFill) {
+			fillPos := p.Pos()
+			_ = p.lexer.consumeToken() // consume FILL
+			fill, err = p.parseFillClause(fillPos)
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			// Not WITH FILL, restore state
+			p.lexer.restoreState(savedState)
+		}
+	}
```

Comment:
Perhaps we can replace this block with `tryConsumeKeywords`?

Reply:
for sure, done